### DISCUSSION TITLE
CNN Training problem

### DISCUSSION THREAD

**Discussion by quinteroossa** [kitzeslab/opensoundscape#984] [Q&A]
Hi all! I hope you are doing well
I am currently having some problems when trying to train a CNN model. I have tried to solve it by changing some network parameters, but is not working. Has someone have this problem?
![Screenshot 2024-04-30 at 1 42 39 PM](https://github.com/kitzeslab/opensoundscape/assets/84196558/b66688c7-104e-4260-835b-2f0f954bde8a)
Thank you for any help in advance!

**Comment by sammlapp** [comment]
Hi @quinteroossa , I'm not familiar with this error, but it would be helpful to see more (or all) of your training notebook to investigate the potential causes. 

The warning message refers to JAX and we don't use JAX in OpenSoundscape, so I wonder if you are creating the CNN model object from some other model, for instance from the Google Perch model? 

If you can share a version of your colab notebook with any private parts redacted, I can take a look

**Comment by sammlapp** [comment]
for reference, we found that the error was caused by using num_workers>0. At least in the free hardware available on Google Colab, multithreading seems to be disabled. Using num_workers=0 should solve this.
